A patch to implement various things to be FHS (and Debian policy)
compliant: PID files will go to /var/run; the configuration files
will go to /etc/cfengine3 (same as we did with cfengine2); logs to
/var/log

That symlink is created by debian/links but we need to remove the
actual creation of the inputs dir any cf-* binary will fail

--- a/src/generic_agent.c
+++ b/src/generic_agent.c
@@ -392,9 +392,10 @@
    CfOut(cf_verbose,"","Work directory is %s\n",CFWORKDIR);
 
    snprintf(HASHDB,CF_BUFSIZE-1,"%s%c%s",CFWORKDIR,FILE_SEPARATOR,CF_CHKDB);
-
+/*
    snprintf(vbuff,CF_BUFSIZE,"%s%cinputs%cupdate.conf",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
    MakeParentDirectory(vbuff,force);
+*/
    snprintf(vbuff,CF_BUFSIZE,"%s%cbin%ccf-agent -D from_cfexecd",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
    MakeParentDirectory(vbuff,force);
    snprintf(vbuff,CF_BUFSIZE,"%s%coutputs%cspooled_reports",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
@@ -1047,7 +1048,7 @@
    FatalError("Uname couldn't get kernel name info!!\n");
    }
 
-snprintf(LOGFILE,CF_BUFSIZE,"%s%ccfagent.%s.log",CFWORKDIR,FILE_SEPARATOR,VSYSNAME.nodename);
+snprintf(LOGFILE,CF_BUFSIZE,"/var/log/cfagent.%s.log",VSYSNAME.nodename);
 VSETUIDLOG = strdup(LOGFILE);
 
 snprintf(vbuff,CF_BUFSIZE,"%s%c.",CFWORKDIR,FILE_SEPARATOR);
@@ -1605,7 +1606,7 @@
 
 { FILE *fp;
 
-snprintf(PIDFILE,CF_BUFSIZE-1,"%s%c%s",CFWORKDIR,FILE_SEPARATOR,filename);
+snprintf(PIDFILE,CF_BUFSIZE-1,"/var/run/%s",filename);
 
 if ((fp = fopen(PIDFILE,"w")) == NULL)
    {
--- a/src/agent.c
+++ b/src/agent.c
@@ -331,7 +331,7 @@
   status which we need for setting returns
 */
 
-snprintf(filename,CF_BUFSIZE,"%s/cfagent.%s.log",CFWORKDIR,VSYSNAME.nodename);
+snprintf(filename,CF_BUFSIZE,"/var/log/cfagent.%s.log",VSYSNAME.nodename);
 MapName(filename);
 if ((fp = fopen(filename,"a")) != NULL)
    {
--- a/src/files_interfaces.c
+++ b/src/files_interfaces.c
@@ -332,7 +332,7 @@
       }
    }
 
-snprintf(filename,CF_BUFSIZE,"%s/cfagent.%s.log",CFWORKDIR,VSYSNAME.nodename);
+snprintf(filename,CF_BUFSIZE,"/var/log/cfagent.%s.log",VSYSNAME.nodename);
 MapName(filename);
 
 if (!LoadFileAsItemList(&VSETUIDLIST,filename,a,pp))
--- a/src/transaction.c
+++ b/src/transaction.c
@@ -156,7 +156,7 @@
    sum = (CF_MACROALPHABET * sum + cc_operand[i]) % CF_HASHTABLESIZE;
    }
 
-snprintf(cflog,CF_BUFSIZE,"%s/cf3.%.40s.runlog",CFWORKDIR,host);
+snprintf(cflog,CF_BUFSIZE,"/var/log/cf3.%.40s.runlog",host);
 snprintf(cflock,CF_BUFSIZE,"lock.%.100s.%s.%.100s_%d_%s",pp->bundle,cc_operator,cc_operand,sum,str_digest);
 snprintf(cflast,CF_BUFSIZE,"last.%.100s.%s.%.100s_%d_%s",pp->bundle,cc_operator,cc_operand,sum,str_digest);
 
